azure_monitor/src/azure_monitor/sdk/auto_collection/live_metrics/__init__.py

```diff
@@ -0,0 +1,151 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+#
+import time
+import typing
+import uuid
+
+from azure_monitor.protocol import BaseObject
+from azure_monitor.utils import azure_monitor_context
+
+LIVE_METRICS_SUBSCRIBED_HEADER = "x-ms-qps-subscribed"
+STREAM_ID = str(uuid.uuid4())
+
+
+def create_metric_envelope(instrumentation_key: str):
+    envelope = LiveMetricEnvelope(
+        documents=None,
+        instance=azure_monitor_context.get("ai.cloud.roleInstance"),
+        instrumentation_key=instrumentation_key,
+        invariant_version=1,  # 1 -> v1 QPS protocol,
+        machine_name=azure_monitor_context.get("ai.device.id"),
+        metrics=None,
+        stream_id=STREAM_ID,
+        timestamp="/Date({0})/".format(str(int(time.time()) * 1000)),
+        version=azure_monitor_context.get("ai.internal.sdkVersion"),
+    )
+    return envelope
+
+
+class LiveMetricDocumentProperty(BaseObject):
+
+    __slots__ = ("key", "value")
+
+    def __init__(self, key: str, value: str) -> None:
+        self.key = key
+        self.value = value
+
+
+class LiveMetricDocument(BaseObject):
```

Comment:
Just curious, where are the specs for these schemas defined? I can't seem to find them here https://github.com/microsoft/ApplicationInsights-Home/tree/master/EndpointSpecs/Schemas/Bond

Reply:
There are some internal specs but you can use .NET classes as reference
https://github.com/microsoft/ApplicationInsights-dotnet/tree/develop/WEB/Src/PerformanceCollector/Perf.Shared/Implementation/QuickPulse/Service%20contract